lib/class-vip-vaultpress-ping-cleanup.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+class VIP_VaultPress_Ping_Cleanup {
+	const VP_PING_OPTION_NAME = '_vp_ai_ping%';
+	const OPTION_NAME         = 'vip_vaultpress_ping_cleanup_complete';
+	const CRON_HOOK           = 'vip_vaultpress_ping_cleanup';
+	const CRON_INTERVAL       = 'hourly';
+
+	// How many options to delete per cron run.
+	const QUERY_LIMIT = 100;
+
+	public static function init(): void {
+		if ( ! class_exists( 'VaultPress' ) && defined( 'ENABLE_VIP_VAULTPRESS_PING_CLEANUP' ) && true === ENABLE_VIP_VAULTPRESS_PING_CLEANUP ) {
+			add_action( 'init', array( __CLASS__, 'schedule_cron' ), 99999 );
+			add_action( self::CRON_HOOK, array( __CLASS__, 'do_cron' ) );
+		}
+	}
+
+	public static function schedule_cron(): void {
+		$completed = get_option( self::OPTION_NAME );
+
+		if ( $completed && wp_next_scheduled( self::CRON_HOOK ) ) {
+			wp_clear_scheduled_hook( self::CRON_HOOK );
+			return;
+		}
+
+		if ( false === $completed && ! wp_next_scheduled( self::CRON_HOOK ) ) {
+			wp_schedule_event( time(), self::CRON_INTERVAL, self::CRON_HOOK );
+		}
+	}
```

Comment:
It would be best to not have to run this code on every request. Rather than managing a separate cron job for this task, what if we instead hooked onto an existing one, maybe `a8c_cron_control_clean_legacy_data`?

Can hook onto the hook and have it check to ensure VaultPress is not currently active. And if not, then delete all the excess legacy VP data. I'm also not so sure we need to store the results in a new option called `vip_vaultpress_ping_cleanup_complete`. If Vaulpress isn't active and `requires_cleanup()` returns false, then we're good.